### PR TITLE
Start texture atlases off small, and allow them to grow.

### LINKED ISSUE #193: moires runs very slow
really amazing demo! I tried cloning servo and compiling it myself, but I'm pretty consistently getting 5 fps on the moire demo (on a pretty recent macbook pro).

 - I compiled with `./mach build --release`
 - I cloned moire.html from https://github.com/pcwalton/webrender-demos
 - I ran servo with `target/release/servo -Z wr-stats -w --resolution=1024x768 --pref dom.mozbrowser.enabled ../webrender-demos/moire.htm` (I copied it straight from the demo stream)

this is what I'm seeing: (note the 4 fps)

Interestingly, this seems to be limited to moires. Both spheres and transparent_rects run at ~60fps.

Any thoughts?

### PR DESCRIPTION
**Pull request by pcwalton**
Includes #199.

Closes #193.

r? @glennw

### CODE REVIEW

File: src/freelist.rs [hunk 1]
```diff
@@ -97,4 +99,9 @@ impl<T: FreeListItem> FreeList<T> {
         item.set_next_free_id(self.first_free_index);
         self.first_free_index = Some(id);
     }
+
+    /// NB: This iterates over free items too!
```
Comment (glennw) [inline src/freelist.rs]: I'm uneasy about this since it exposes internal details - is there a way to avoid it?

Comment (pcwalton) [inline src/freelist.rs]: Maybe I can yield indices instead—let me see.

**Comment by pcwalton** [comment]
Fixed. r? @glennw

**Comment by frewsxcv** [comment]
Just published heapsize_plugin 0.1.4, which *should* cause Travis to pass again.

**Pull request merged**